### PR TITLE
revert old timer in subscribe(channel_service) for compatibility

### PR DESCRIPTION
**Pull request by laurenceyoon**
Supports backward compatibility so that citizen node can subscribe continuously even it's disconnected from previous versions of the network.

### CODE REVIEW

File: loopchain/channel/channel_service.py [hunk 1]
```diff
@@ -497,20 +497,90 @@ def connect_to_radio_station(self, is_reconnect=False):
...
-            peer_target = ChannelProperty().rest_target
+        response = {'response_code': message_code.Response.fail,
+                    'message': message_code.get_response_msg(message_code.Response.fail)}
+
+        try:
+            if conf.REST_SSL_TYPE == conf.SSLAuthType.none:
+                peer_target = ChannelProperty().rest_target
+            else:
+                peer_target = f"https://{ChannelProperty().rest_target}"
+            response = rs_rest_stub.call(
```
Comment (zsaladin) [inline loopchain/channel/channel_service.py]: `rs_rest_stub.call()` could be blocked.
I recommnend replacing it with `await rs_rest_stub.call_async()`

File: loopchain/channel/channel_service.py [hunk 2]
```diff
@@ -497,20 +497,90 @@ def connect_to_radio_station(self, is_reconnect=False):
...
+                        target=timer_key,
+                        duration=conf.SHUTDOWN_TIMER,
+                        callback=self.shutdown_peer,
+                        callback_kwargs={"message": error}
+                    )
+                )
+
+        return response
+
+    def __check_block_height_call_to_rs_stub(self, **kwargs):
```
Comment (zsaladin) [inline loopchain/channel/channel_service.py]: def -> async def 
Then you can use `rest_stub.call_async`

File: loopchain/channel/channel_service.py [hunk 3]
```diff
@@ -497,20 +497,90 @@ def connect_to_radio_station(self, is_reconnect=False):
...
+        rs_rest_stub = kwargs.get("rs_rest_stub", None)
+        try:
+            last_block = rs_rest_stub.call("GetLastBlock")
+            if last_block['height'] <= self.__block_manager.get_blockchain().block_height:
+                return
+            else:
+                # citizen needs additional block or connected to problematic peer.
+                raise Exception
```
Comment (zsaladin) [inline loopchain/channel/channel_service.py]: This exception does not seem useful.
Why don't you remove `try, except` and `else`

**Review by zsaladin — APPROVED** [review]

File: loopchain/channel/channel_service.py (hunk 1 above)
Comment (laurenceyoon) [reply loopchain/channel/channel_service.py]: Updated to async call

File: loopchain/channel/channel_service.py (hunk 2 above)
Comment (laurenceyoon) [reply loopchain/channel/channel_service.py]: updated to call_async and changed method to async.

File: loopchain/channel/channel_service.py [hunk 4]
```diff
@@ -501,16 +501,79 @@ def connect_to_radio_station(self, is_reconnect=False):
...
+        if last_block['height'] <= self.__block_manager.get_blockchain().block_height:
+            return
+
+        timer_key = TimerService.TIMER_KEY_GET_LAST_BLOCK_KEEP_CITIZEN_SUBSCRIPTION
+        if timer_key in self.__timer_service.timer_list.keys():
+            util.logger.spam(f"stop timer for check_block_height_call to radiostation...")
+            self.__timer_service.stop_timer(timer_key)
+
+        timer_key = TimerService.TIMER_KEY_SUBSCRIBE
+        if timer_key not in self.__timer_service.timer_list.keys():
```
Comment (yakkle) [inline loopchain/channel/channel_service.py]:
```suggestion
        if timer_key not in self.__timer_service.timer_list:
```
to check key in dict, don't need get dict.keys().

Comment (laurenceyoon) [reply loopchain/channel/channel_service.py]: Updated.

**Review by yakkle — APPROVED** [review]

**Pull request merged**